### PR TITLE
Various updates

### PR DESCRIPTION
**Pull request by petersteneteg**
Updates of: GLM, PyBind11, GTest, Warn, Lots of warning fixes. Some cleanup of pybind glm bindings.

### CODE REVIEW

File: ext/glm/CMakeLists.txt [hunk 1]
```diff
@@ -1,19 +1,22 @@
...
     GLM_ENABLE_EXPERIMENTAL
+    GLM_FORCE_UNRESTRICTED_GENTYPE
+    GLM_FORCE_SILENT_WARNINGS
 )
 
 if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang")
     # Needed when using clang since GLM fails to detect c++11/14 support
     # https://github.com/g-truc/glm/issues/620
-    target_compile_definitions(glm INTERFACE GLM_FORCE_CXX14)
+    #target_compile_definitions(glm INTERFACE GLM_FORCE_CXX14)
```
Comment (ResearchDaniel) [inline ext/glm/CMakeLists.txt]: This if statement can be removed now?

File: ext/glm/CMakeLists.txt [hunk 2]
```diff
@@ -1,19 +1,22 @@
...
 endif()
 
-set_target_properties(glm_dummy PROPERTIES VERSION 0.9.9)
-ivw_folder(glm_dummy ext)
+#set_target_properties(glm_shared PROPERTIES VERSION ${GLM_VERSION})
```
Comment (ResearchDaniel) [inline ext/glm/CMakeLists.txt]: Decomment?

File: ext/gtest/CMakeLists.txt [hunk 3]
```diff
@@ -3,9 +3,8 @@ if(IVW_UNITTESTS OR IVW_INTEGRATION_TESTS)
     # settings on Windows
     set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
 
-    # Add googletest directly to our build. This defines
-    # the gtest and gtest_main targets.
-    add_subdirectory(gtest/googletest)
+    option(BUILD_GMOCK "Builds the googlemock subproject" OFF)
```
Comment (ResearchDaniel) [inline ext/gtest/CMakeLists.txt:6]: Googlemock? Why explicitly disable?

File: ext/gtest/CMakeLists.txt [hunk 4]
```diff
@@ -15,11 +14,4 @@ if(IVW_UNITTESTS OR IVW_INTEGRATION_TESTS)
         target_compile_definitions(gtest INTERFACE -DGTEST_LINKED_AS_SHARED_LIBRARY=1)
         #target_compile_definitions(gtest_main PRIVATE -DGTEST_LINKED_AS_SHARED_LIBRARY=0)
     endif()
-
-    ivw_register_license_file(NAME "Google C++ Testing Framework" TARGET gtest MODULE Core
```
Comment (ResearchDaniel) [inline ext/gtest/CMakeLists.txt:19]: Is this added somewhere else?

File: src/core/CMakeLists.txt [hunk 5]
```diff
@@ -697,7 +697,13 @@ ivw_register_license_file(NAME "Flags" MODULE Core
     FILES ${IVW_EXTENSIONS_DIR}/flags/LICENSE
 )
 
-ivw_register_license_file(NAME "OpenGL Mathematics (GLM)" TARGET glm_dummy MODULE Core
+ivw_register_license_file(NAME "{fmt}" target fmt MODULE Core
```
Comment (ResearchDaniel) [inline src/core/CMakeLists.txt:700]: {fmt} - >${fmt}

**Review by ResearchDaniel — CHANGES_REQUESTED** [review]
Looks good! Some minor changes required

File: ext/gtest/CMakeLists.txt (hunk 3 above)
Comment (petersteneteg) [reply ext/gtest/CMakeLists.txt:6]: We are not using it...

File: ext/gtest/CMakeLists.txt (hunk 4 above)
Comment (petersteneteg) [reply ext/gtest/CMakeLists.txt:19]: Yes, in core, unfortunatly it does not work in that spot...

File: src/core/CMakeLists.txt (hunk 5 above)
Comment (petersteneteg) [reply src/core/CMakeLists.txt:700]: Ahh no... thats its name.

File: ext/glm/CMakeLists.txt (hunk 1 above)
Comment (petersteneteg) [reply ext/glm/CMakeLists.txt]: Yes, will do..

File: ext/glm/CMakeLists.txt (hunk 2 above)
Comment (petersteneteg) [reply ext/glm/CMakeLists.txt]: I add it manually to get the patch version also...

**Review by ResearchDaniel — APPROVED** [review]

**Pull request merged**